src/network/networkd-address-generation.c

```diff
@@ -370,16 +370,11 @@ int config_parse_address_generation_type(
                 }
 
                 if (comma) {
-                        r = sd_id128_from_string(comma + 1, &secret_key);
+                        r = id128_from_string_nonzero(comma + 1, &secret_key);
                         if (r < 0) {
                                 log_syntax(unit, LOG_WARNING, filename, line, r,
-                                           "Failed to parse secret key in %s=, ignoring assignment: %s",
-                                           lvalue, rvalue);
-                                return 0;
-                        }
-                        if (sd_id128_is_null(secret_key)) {
-                                log_syntax(unit, LOG_WARNING, filename, line, 0,
-                                           "Secret key in %s= cannot be null, ignoring assignment: %s",
+                                           r == -ENXIO ? "Secret key in %s= cannot be null, ignoring assignment: %s"
+                                                       : "Failed to parse secret key in %s=, ignoring assignment: %s",
```

Comment:
> In config_parse_address_generation_type()
we would set the output parameter and then say it's ignored, so it looks
like an error, but the previous value is SD_ID128_NULL too, so the code was
fine.

I cannot follow the commit message... What kind of error do you fix?

Reply:
You snipped the first part of the message that specified that there is no functional change. It's not fixing an error, it is just making the code shorter and clearer.

Reply:
Still I cannot follow the commit message, but the code looks fine.